### PR TITLE
TA#21559 Stock component shadow moves

### PR DESCRIPTION
**Pull request by ddufresne**

### CODE REVIEW

File: stock_serial_single_quant/models/stock_move.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline stock_serial_single_quant/models/stock_move.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5278552)

File: stock_serial_single_quant/models/stock_move.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline stock_serial_single_quant/models/stock_move.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5278552)

File: stock_component/models/stock_move.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline stock_component/models/stock_move.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5278552)

File: stock_component/models/stock_move.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline stock_component/models/stock_move.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5278552)

File: stock_serial_single_quant/models/stock_move_line.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline stock_serial_single_quant/models/stock_move_line.py]: Codacy Issue found: [Unused fields imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5278552)

File: stock_component/models/stock_move.py (hunk 3 above)
Comment (foutoucour) [inline stock_component/models/stock_move.py]:
```suggestion
from odoo import api, fields, models
```

File: stock_component/models/stock_move.py (hunk 4 above)
Comment (foutoucour) [inline stock_component/models/stock_move.py]:
```suggestion
```

File: stock_serial_single_quant/models/stock_move.py (hunk 2 above)
Comment (foutoucour) [inline stock_serial_single_quant/models/stock_move.py]:
```suggestion
from odoo import api, fields, models
```

File: stock_serial_single_quant/models/stock_move.py (hunk 1 above)
Comment (foutoucour) [inline stock_serial_single_quant/models/stock_move.py]:
```suggestion
```

File: stock_serial_single_quant/models/stock_move_line.py (hunk 5 above)
Comment (foutoucour) [inline stock_serial_single_quant/models/stock_move_line.py]:
```suggestion
from odoo import api, models, _
```

File: stock_component/models/stock_production_lot.py [hunk 6]
```diff
@@ -37,9 +35,13 @@ def _get_parent_component(self):
             [("component_line_ids.component_id", "=", self.id)], limit=1
         )
 
+    def pull_components(self):
```
Comment (foutoucour) [inline stock_component/models/stock_production_lot.py:38]: pourquoi `pull_components` est publique mais pas `_pull_component` ? pour éviter une confusion?

File: stock_component/models/stock_move.py (hunk 3 above)
Comment (ddufresne) [inline stock_component/models/stock_move.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5278552)

File: stock_component/models/stock_move.py (hunk 4 above)
Comment (ddufresne) [inline stock_component/models/stock_move.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5278552)

File: stock_component/models/stock_production_lot.py [hunk 7]
```diff
@@ -86,7 +86,7 @@ def _check_has_no_parent_component(self):
             )
 
     def _check_has_one_non_null_quant(self):
-        quants = self.get_non_zero_quants()
+        quants = self.get_positive_quants()
```
Comment (foutoucour) [inline stock_component/models/stock_production_lot.py:89]: :+1: pour le nouveau nom de la fonction

File: stock_component/models/stock_production_lot.py (hunk 6 above)
Comment (ddufresne) [reply stock_component/models/stock_production_lot.py:38]: @foutoucour pour rendre l'intention explicite. Toutes les composantes devraient toujours être pullées en même temps. Il n'y a pas de raison pour du code externe de vouloir puller une composante en particulier.

**Review by foutoucour — APPROVED** [review]

**Pull request merged**